android-generator/src/test/scala/models/generator/android/TestAndroidClasses.scala

```diff
@@ -130,12 +132,23 @@ class TestAndroidClasses
     val result = AndroidClasses.invoke(InvocationForm(models.TestHelper.service(json.format())))
 
     result.isRight should be(true)
-    result.right.get.size should be(4)
-    result.right.get(0).name should be("CarType.java")
-    result.right.get(1).name should be("Model.java")
-    result.right.get(2).name should be("ApidocObjectMapper.java")
-    result.right.get(3).name should be("ModelsClient.java")
-
+    val files = result.right.get
+    files.size should be(4)
+    files(0).name should be("CarType.java")
+    files(1).name should be("Model.java")
+    files(2).name should be("ApidocObjectMapper.java")
+    files(3).name should be("ModelsClient.java")
+    files.foreach {
+      assertValidJavaSourceFile(_)
+    }
   }
 
+  def assertValidJavaSourceFile(sourceFile: com.bryzek.apidoc.generator.v0.models.File): Unit = {
+    sourceFile.name.endsWith(".java") shouldBe true
```

Comment:
Do we need this check, given we're validating the file names - e.g., `files(0).name should be ("CarType.java")`.